pkg/ocm/sca/sca.go

```diff
@@ -219,6 +222,11 @@ func (c *Controller) requestSCAWithExpBackoff(endpoint string) ([]byte, error) {
 	var data []byte
 	err := wait.ExponentialBackoff(bo, func() (bool, error) {
 		var err error
+		ctx := context.TODO()
+		_, err = gatherArchitectures(ctx, c.gatherProtoKubeConfig)
+		if err != nil {
+			return true, err
+		}
```

Comment:
Note that the `ExponentialBackoff` function can be called multiple times and I think we don't need to gather the architectures repeatedly in this case. So I would suggest to move this logic out of `requestSCAWithExpBackoff` and it should also help with the `TODO` context usage.

Reply:
You are right. Moved out of the body of the inner function